### PR TITLE
Support GHC 9.8

### PR DESCRIPTION
**Pull request by amesgen**

### CODE REVIEW

File: vector-th-unbox.cabal [hunk 1]
```diff
@@ -17,10 +17,17 @@ category:       Data
...
+  GHC == 8.2,
+  GHC == 8.4,
+  GHC == 8.6,
+  GHC == 8.8,
+  GHC == 8.10,
+  GHC == 9.0,
+  GHC == 9.2,
+  GHC == 9.4,
+  GHC == 9.6,
+  GHC == 9.8
```
Comment (andreasabel) [inline vector-th-unbox.cabal]: Is there a pretext for this form of constraint (`GHC == x.y`)?  E.g. I am not sure `haskell-ci` would understand it.  
I have seen `GHC == 8.0.*` before, meaning `GHC == { 8.0.1, 8.0.2 }`.
Alas, `cabal check` does not check these constraints, so there is wild growth...

**Review by andreasabel — APPROVED** [review]

File: vector-th-unbox.cabal (hunk 1 above)
Comment (amesgen) [reply vector-th-unbox.cabal]: Ah, good point, I just wanted to make it consistent with the specification in the GHA `haskell.yml`. Happy to add back the third component.

According to https://hackage-search.serokell.io/?q=GHC+%3D%3D+.%5C...%3F%28%2C%7C%24%29, there are already lots of packages on Hackage that use this style:

> Search completed (238 matches across 90 packages)

Comment (andreasabel) [reply vector-th-unbox.cabal]: Yeah, suppose this can be considered a kind of "free-form" field for most cases.

**Comment by andreasabel** [comment]
Published this as revision 5: https://hackage.haskell.org/package/vector-th-unbox-0.2.2/revisions/

Btw, @amesgen, I prefer to list newest GHCs first in the `tested-with`, because if you look at the revision report (link above) you see the most important information first.

**Comment by amesgen** [comment]
> Btw, @amesgen, I prefer to list newest GHCs first in the `tested-with`, because if you look at the revision report (link above) you see the most important information first.

Nice, changed that :+1:

**Review by fumieval — APPROVED** [review]

**Comment by andreasabel** [comment]
@fumieval: Any obstacles to merge this?

(We are already entering the next lap, GHC 9.10.)

**Pull request merged**